streams/src/test/java/org/apache/kafka/streams/processor/TopologyBuilderTest.java

```diff
@@ -227,25 +227,27 @@ public void testAddStateStoreWithDuplicates() {
     @Test
     public void testAddStateStore() {
         final TopologyBuilder builder = new TopologyBuilder();
-        List<StateStoreSupplier> suppliers;
 
         StateStoreSupplier supplier = new MockStateStoreSupplier("store-1", false);
         builder.addStateStore(supplier);
-        suppliers = builder.build("X", null).stateStoreSuppliers();
-        assertEquals(0, suppliers.size());
+        builder.setApplicationId("X");
+
+        assertEquals(0, builder.build(null).stateStoreSuppliers().size());
 
         builder.addSource("source-1", "topic-1");
         builder.addProcessor("processor-1", new MockProcessorSupplier(), "source-1");
         builder.connectProcessorAndStateStores("processor-1", "store-1");
-        suppliers = builder.build("X", null).stateStoreSuppliers();
+
+        List<StateStoreSupplier> suppliers = builder.build(null).stateStoreSuppliers();
         assertEquals(1, suppliers.size());
         assertEquals(supplier.name(), suppliers.get(0).name());
     }
 
     @Test
     public void testTopicGroups() {
         final TopologyBuilder builder = new TopologyBuilder();
-
+        builder.setApplicationId("X");
+        builder.addInternalTopic("topic-1x");
```

Comment:
I was basically worried about naming conflicts where a user can add a source topic that happens to match the topic name used for a temporary topic (I guess an internal topic used for shuffles). From the test it looked like we were keeping the set of source topics, but the source topic for an internal topic was maintained in that set without the prefix (i.e. it isn't actually the full topic name).

I think this is an existing issue since this was just a refactoring. But it seems like we should track these topics such that a conflict could be caught. It seemed like with the current setup we'd just sort of silently accept the conflict. (Or maybe I am just getting confused about the behavior as I'm not very familiar with the code -- I can try to write a test case that causes the problem I am imagining.)